### PR TITLE
xgboost update

### PR DESCRIPTION
**Pull request by double-y**
### Description
Added na.action argument so that users can decide how to handle NA values

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [x] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/build_model.R [hunk 1]
```diff
@@ -41,43 +41,47 @@ build_model_ <- function(data, model_func, seed = 0, test_rate = 0, group_cols =
     stop("test_rate has to be between 0 and 1")
   }
 
+  # use this for preprocess
+  processed <- data
```
Comment (hideaki) [inline R/build_model.R]: why is this change necessary? some more explanation would help.

File: R/build_xgboost.R [hunk 2]
```diff
@@ -1,12 +1,27 @@
 #' formula version of xgboost
 #' @export
```
Comment (hideaki) [inline R/build_xgboost.R]: Explanation of na.action parameter would help here.

File: R/build_model.R [hunk 3]
```diff
@@ -41,43 +41,47 @@ build_model_ <- function(data, model_func, seed = 0, test_rate = 0, group_cols =
...
+  group_col_names <- grouped_by(processed)
 
   # integrate all additional parameters
   dots <- lazyeval::all_dots(.dots, ...)
 
   # check if variables in grouped_col_names are not used
   formula <- dots$formula
   if(!is.null(formula)){
-    vars <- all.vars(formula$expr)
+    # expand formula with column names of data
```
Comment (hideaki) [inline R/build_model.R]: is this change related to na.action, or is it for something else?

File: R/build_model.R (hunk 1 above)
Comment (double-y) [inline R/build_model.R]: Ok, I'll add a comment

File: R/build_model.R (hunk 3 above)
Comment (double-y) [inline R/build_model.R]: Ah, this is for a different issue. dot was not handled well, so uses terms function to expand it.

File: R/build_xgboost.R (hunk 2 above)
Comment (double-y) [inline R/build_xgboost.R]: Ok, will add it

File: R/build_model.R (hunk 3 above)
Comment (hideaki) [inline R/build_model.R]: Please explain that in a comment so that we accumulate the knowledge of why we do it certain way.
